src/vs/workbench/browser/layout.ts

```diff
@@ -539,6 +539,48 @@ export class WorkbenchLayout implements IVerticalSashLayoutProvider, IHorizontal
 		return this.panelMaximized;
 	}
 
+	// Add reset for keyboard support
+	public resetSideBarWidth(): number {
+		this.sidebarWidth = MIN_SIDEBAR_PART_WIDTH;
+		return this.sidebarWidth;
+	}
+
+	// change part size along the main axis
+	public setPartSizeChange(part: Parts, sizeChange: number): boolean {
```

Comment:
This method should do the same validation about minimal and maximal sizes for a specific part as we already do when the user resizes a view using the sashes (e.g. [here](https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/browser/layout.ts#L192))